### PR TITLE
Make environment files management consistent between `pip` and `conda`

### LINKED ISSUE #230: Add automated approach to validate that requirement files are consistent
**Is your feature request related to a problem? Please describe.**
As mentioned in #229, there are currently four places where dependencies are specified:

- environment.yml
- setup.py
- dev-requirements.txt
- [meta.yaml](https://github.com/conda-forge/geoutils-feedstock/blob/master/recipe/meta.yaml)

There is a large risk that someone may forget to update all of these if a dependency is updated, so there should be an automated way to validate that they are consistent.

**Describe the solution you'd like**
A test or pre-commit hook to validate the content of all files.

Some pseudocode:
```python
min_python_version = "3.7"
dependencies = ["numpy>=1.17", "scipy", ...]  # This could either be hardcoded in the test or read from a "reference" reqs file.
dev_dependencies = ["pytest", "sphinx"]

for req_file in req_files:
    # Standardize -/_ (e.g. sphinxcontrib_programoutput on PyPi and sphinxcontrib-programoutput on conda-forge)
    # Validate that all dependencies (and dev-dependencies if appropriate) are there
    # Validate the version constraints
    # Check that no additional dependencies exist that are not defined above.
```
This should fail with a descriptive error if something is inconsistent.

**Describe alternatives you've considered**
Perhaps some bot exists to do this for us?

For reference, Pandas has a pre-commit hook for this specifically.

### LINKED ISSUE #373: Force `geopandas > 0.12.0` to support `normalize`
See https://github.com/GlacioHack/xdem/issues/360#issuecomment-1527735925

### LINKED ISSUE #397: Setup a PR template to ensure `setup.py` changes are properly reported
Mirrorring https://github.com/GlacioHack/xdem/issues/425

### PR DESCRIPTION
**Pull request by rhugonnet**
Mirroring https://github.com/GlacioHack/xdem/pull/429

Resolves #397
Resolves #230
Resolves #373

Also fixes a minor floating precision bug in `test_vector`

### CODE REVIEW

File: doc/source/how_to_install.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ Once installed, the same commands can be run by simply replacing `conda` by `mam
 pip install geoutils
 ```
 
-```{note}
-Setting up GDAL and PROJ may require some extra steps, depending on your operating system and configuration.
+```{warning}
+Updating packages with `pip` (and sometimes `mamba`) can break your installation. If this happens, re-create an environment from scratch fixing directly all your dependencies.
```
Comment (adehecq) [inline doc/source/how_to_install.md:26]: I think "setting" or "defining" would be more appropriate than "fixing" (Frenglish?).

File: HOW_TO_RELEASE.md [hunk 2]
```diff
@@ -18,7 +20,7 @@ An automatic GitHub action will start to push and publish the new release to PyP
 - Before releasing, you need to delete **both** the tag and the release of the previous release. If you release with the same tag without deletion, it will ignore your commit changing the version number, and PyPI will block the upload again. You're stuck in a circle.
 
 
-## The hard way
+### The hard way
 
  1. Go to your local main repository (not the fork) and ensure your master branch is synced:
        git checkout master
```
Comment (adehecq) [inline HOW_TO_RELEASE.md:26]: actually, master does not exist anymore, it should be "main".

File: doc/source/how_to_install.md (hunk 1 above)
Comment (rhugonnet) [reply doc/source/how_to_install.md:26]: Good point, I meant "pinning versions for your other dependencies", changed in both #401 and https://github.com/GlacioHack/xdem/pull/433

File: HOW_TO_RELEASE.md (hunk 2 above)
Comment (rhugonnet) [reply HOW_TO_RELEASE.md:26]: Also done!

**Comment by rhugonnet** [comment]
Thanks for the comments!
It was a lot of reading, #230 summarizes things well! A good thing taken care of :smile:

**Pull request merged**